lib/temporal/configuration.rb

```diff
@@ -6,8 +6,11 @@ class Configuration
     attr_reader :timeouts
     attr_accessor :client_type, :host, :port, :logger, :metrics_adapter, :namespace, :task_queue, :headers
 
+    # We want an infinite execution timeout for cron schedules and other perpetual workflows.
+    # We choose an 10-year execution timeout because that's the maximum the cassandra DB supports,
+    # matching the go SDK, see https://github.com/temporalio/sdk-go/blob/d96130dad3d2bc189bc7626543bd5911cc07ff6d/internal/internal_workflow_testsuite.go#L68
```

Comment:
nit: Probably not the best reference since it's coming from a test suite and not production code